crates/ruma-common/src/push/condition.rs

```diff
@@ -254,7 +254,7 @@ pub struct PushConditionRoomCtx {
 
 /// The room power levels context to be able to test the corresponding push conditions.
```

Comment:
`PushConditionRoomCtx` needs to be non-exhaustive too.

It even has a field behind a cargo feature, which makes the feature non-additive.

However this type will need a constructor that populates the first 4 (required) fields.

Reply:
Ah, maybe this is the type I was thinking of being constructed directly in the SDK? (outside of tests)

Reply:
Yes probably, the other is almost always constructed from the power levels (except that one time in a test, where it could be constructed from the default power levels).